src/OSPSuite.Infrastructure.Import/Services/IImporter.cs

```diff
@@ -119,6 +120,14 @@ IEnumerable<MetaDataMappingConverter> mappings
          })).ToList();
       }
 
+      public IReadOnlyList<IReadOnlyDictionary<string, string>> EnumerateMetaData(Cache<string, IDataSet> dataSets, IEnumerable<MetaDataMappingConverter> mappings)
```

Comment:
I don't understand this return structure. Why is this so complicated?

Reply:
comment still applies here. The fact that were are passing such primitive values in and out results in indexes being pass along everywhere. This may work, but it is very complicated to reason about